packages/ui/src/Popover/index.tsx

```diff
@@ -0,0 +1,158 @@
+
+/**
+ * A popover box that appears next to the trigger element.
+ *
+ * To render a popover, compose it using a few components: `<Popover />`,
+ * `<Popover.Trigger />`, and `<Popover.Content />`. The latter two must be
+ * descendents of `<Popover />` in the component tree, and siblings to each
+ * other. (`<Popover.Trigger />` is optional, though — more on that in a moment.)
+ *
+ * ```tsx
+ * <Popover>
+ *   <Popover.Trigger asChild>
+ *     <Button>Open the popover</Button>
+ *   </Popover.Trigger>
+ *
+ *   <Popover.Content title="Popover title">Popover content here</Popover.Content>
+ * </Popover>
+ * ```
+ *
+ * Depending on your use case, you may want to use `<Popover />` either as a
+ * controlled component, or as an uncontrolled component.
+ *
+ * ## Usage as a controlled component
+ *
+ * Use `<Popover />` as a controlled component when you want to control its
+ * open/closed state yourself (e.g., via a state management solution like
+ * Zustand or Redux). You can accomplish this by passing `isOpen` and `onClose`
+ * props to the `<Popover />` component, and omitting `<Popover.Trigger />`:
+ *
+ * ```tsx
+ * <Button onClick={() => setIsOpen(true)}>Open popover</Button>
+ *
+ * <Popover isOpen={isOpen} onClose={() => setIsOpen(false)}>
+ *   <Popover.Content title="Popover title">Popover content here</Popover.Content>
+ * </Popover>
+ * ```
+ *
+ * Note that, in the example above, the `<Button />` lives outside of the
+ * `<Popover />`, and there is no `<Popover.Trigger />` component rendered inside
+ * the `<Popover />`.
+ *
+ * ## Usage as an uncontrolled component
+ *
+ * If you want to render `<Popover />` as an uncontrolled component, don't pass
+ * `isOpen` or `onClose` to `<Popover />`, and make sure to include a
+ * `<Popover.Trigger />` component inside the `<Popover />`:
+
+ * ```tsx
+ * <Popover>
+ *   <Popover.Trigger asChild>
+ *     <Button>Open the popover</Button>
+ *   </Popover.Trigger>
+ *
+ *   <Popover.Content title="Popover title">Popover content here</Popover.Content>
+ * </Popover>
+ * ```
+ */
+export const Popover = ({ children, onClose, isOpen }: PopoverProps) => {
+  return (
+    <RadixPopover.Root open={isOpen} onOpenChange={value => onClose && !value && onClose()}>
+      {children}
+    </RadixPopover.Root>
+  );
+};
+
+export interface PopoverTriggerProps {
+  children: ReactNode;
+  /**
+   * Change the default rendered element for the one passed as a child, merging
+   * their props and behavior.
+   *
+   * Uses Radix UI's `asChild` prop under the hood.
+   *
+   * @see https://www.radix-ui.com/primitives/docs/guides/composition
+   */
+  asChild?: boolean;
+}
+
+const Trigger = ({ children, asChild }: PopoverTriggerProps) => (
+  <RadixPopover.Trigger asChild={asChild}>{children}</RadixPopover.Trigger>
+);
+Popover.Trigger = Trigger;
+
+export interface PopoverContentProps {
+  children?: ReactNode;
+  side?: RadixPopoverContentProps['side'];
+  align?: RadixPopoverContentProps['align'];
```

Comment:
In this case, I couldn't figure out how to control the position on our side. Sam wrote in Figma that it should open "on the left side if above the trigger" and "on the right side if below the trigger". Radix doesn't allow to set it dynamically - we can choose only one option. So that's why I decided to give this level of control